pkg/tsdb/postgres/postgres_test.go

```diff
@@ -15,12 +15,85 @@ import (
 	"github.com/grafana/grafana/pkg/services/sqlstore/sqlutil"
 	"github.com/grafana/grafana/pkg/tsdb"
 	"github.com/grafana/grafana/pkg/tsdb/sqleng"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 	"xorm.io/xorm"
 
 	_ "github.com/lib/pq"
 	. "github.com/smartystreets/goconvey/convey"
 )
 
+// Test newPostgresQueryEndpoint.
+func TestNewPostgresQueryEndpoint(t *testing.T) {
```

Comment:
Nit. This is fine, but could be simplified by testing the `generateConnectionString` function instead. The test of `newPostgresQueryEndpoint` is covered in integration tests.

Reply:
I chose to test newPostgresQueryEndpoint on purpose, since by depending less on implementation details in tests, you ease the maintenance burden and cognitive load. While it may be easier to write tests for `generateConnectionString`, I would argue that maintenance will be more difficult by tying up implementation details in the test.

Reply:
I think this is more fragile since it depends on a mock. Since both functions take the same argument and `generateConnectionString` is smaller and very specific I think testing that is preferred.

Reply:
@bergquist I've done as you said and made the test for generateConnectionString instead. I think it's problematic however and would rather have kept testing newPostgresQueryEndpoint, since generateConnectionString is only a helper and:

* You lose test coverage of newPostgresQueryEndpoint
* The test becomes more fragile since it depends on the implementation detail of generateConnectionString being used
* Implementation becomes tied up to having the helper function generateConnectionString